site/source/tutorials/007_theming/index.md

```diff
@@ -0,0 +1,136 @@
+
+{% include_codefile 'demo/finished/biz-e-corp/src/main.ts' %}
+
+If you open up your web browser now you should see that the `worker` backgrounds are now `red`.
+
+{% instruction 'Use variables and complete the worker theme' %}
+
+The Dojo 2 build system allows us new `css` features such as `css-custom-properties` by using `postCss` to process our `.m.css` files. We can use these to add variables to `worker.m.css` and to complete it's theme.
+Let go ahead and create `themes/dojo/variables.css` (notice that this file does not have a `.m.css` extension as it is not a `css-module` file).
+
+{% include_codefile 'demo/finished/biz-e-corp/src/themes/dojo/variables.css' lang:css %}
+
+We can now use these variables in `worker.m.css` to create our fully themed `worker`.
+
+{% include_codefile 'demo/finished/biz-e-corp/src/themes/dojo/worker.m.css' lang:css %}
+
+{% section %}
+
+## Theming Dojo widgets
+
+So far in this tutorial we have themed our custom `worker` widget, but how do we target dojo widgets with our theme. To demonstrate this, we will theme the `workerForm` widget as it contains both domNodes and dojo widgets to be themed.
+
+{% instruction 'Lets create `workerForm.m.css` and include it in our theme.ts' %}
+
+{% include_codefile 'demo/finished/biz-e-corp/src/themes/dojo/workerForm.m.css' lang:css %}
+
+{% include_codefile 'demo/finished/biz-e-corp/src/themes/dojo/theme.ts' lines:3,5,9,11,13,16 %}
+
+This should be familiar to you by now as we covered theming `worker` in the last section. Now to theme the Dojo 2 `TextInput` that `WorkerForm` uses, we will need to create a `dojoTextInput.m.css` file and export it from `theme.ts` using a special theme key which is prefixed with `dojo-`. This prefix helps to ensure that our own widget theme files do not clash with dojo specific widgets and makes it easy to recognise when we are targeting one of the other.
+
+{% instruction 'Create `dojoTextInput.m.css` and emport it from `theme.ts`' %}
+
+{% include_codefile 'demo/finished/biz-e-corp/src/themes/dojo/dojoTextInput.m.css' lang:css %}
+
+{% include_codefile 'demo/finished/biz-e-corp/src/themes/dojo/theme.ts' lines:3,5,6,9,11,13,14,16 %}
+
+In this example we are introducing another powerful part of the Dojo 2 theming system, `composes`. Composes comes from the `css-module` specification and allows you to take styles from one class selector and apply them to another. In this case we are saying that `root` of a `TextInput` (the label text in this case), should look the same as the `nameLabel` class in our `WidgetForm`. This approach can be very useful when creating multiple themes from a baseTheme and avoids re-writing / re-specifying styles in multiple places.
```

Comment:
Composes comes from the `css-module` specification and allows you to take styles from one class selector and apply them to another. -> Composes originates in the `css-module` specification, allowing you to apply styles from one class selector to another.